src/librustc/hir/intravisit.rs

```diff
@@ -723,8 +723,8 @@ pub fn walk_pat<'v, V: Visitor<'v>>(visitor: &mut V, pattern: &'v Pat) {
         }
         PatKind::Lit(ref expression) => visitor.visit_expr(expression),
         PatKind::Range(ref lower_bound, ref upper_bound, _) => {
-            visitor.visit_expr(lower_bound);
-            visitor.visit_expr(upper_bound)
+            walk_list!(visitor, visit_expr, lower_bound);
```

Comment:
Yea, it's used in most places in this file where `Option`s are visited. Ostensibly it could be `walk_opt!(...)` instead, but I don't think that should be changed here. `if let` feels like it would contribute a lot of line-noise.